### PR TITLE
Document named zone support for timestamp with time zone

### PR DESCRIPTION
**Pull request by jhlodin**
## Description

Describe how a named time zone is supported by the `timestamp(p) with time zone` type with an example

> Is this change a fix, improvement, new feature, refactoring, or other?

Improvement

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Type documentation

> How would you describe this change to a non-technical end user or system administrator?

Describe an additional supported use of Trino types

## Related issues, pull requests, and links

Follow up on discussion in https://github.com/trinodb/trino/pull/7912#discussion_r958645602

## Documentation

( ) No documentation is needed.
(x) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/src/main/sphinx/language/types.rst [hunk 1]
```diff
@@ -239,11 +239,13 @@ The following examples illustrate the behavior::
 Instant in time that includes the date and time of day with ``P`` digits of
 precision for the fraction of seconds and with a time zone. Values of this
 type are rendered using the time zone from the value.
-Time zones are expressed as the numeric UTC offset value::
+Time zones are expressed as the numeric UTC offset value or a full name::
```
Comment (mosabua) [inline docs/src/main/sphinx/language/types.rst:242]: Should we detail what "full name" means and what standard list is used? Maybe link to it?

**Review by mosabua — CHANGES_REQUESTED** [review]
I think we should also add to TIME WITH TIME ZONE .. and explicitly mention that names are not supported (and why...)

File: docs/src/main/sphinx/language/types.rst (hunk 1 above)
Comment (findepi) [reply docs/src/main/sphinx/language/types.rst:242]: yes!

ours are IANA zone names. there are other standards of zone naming

Comment (jhlodin) [reply docs/src/main/sphinx/language/types.rst:242]: Ahhh it's IANA! I tried to figure out which acronym consortium was responsible for time zones but conveniently forgot about them. Will add that.

**Review by hashhar — APPROVED** [review]
Thanks.

**Comment by hashhar** [comment]
Note that sphinx is complaining about the link -
```
Warning, treated as error:
/docs/src/main/sphinx/language/types.rst:239:unknown document: https://www.iana.org/time-zones
```

File: docs/src/main/sphinx/language/types.rst [hunk 2]
```diff
@@ -239,11 +239,14 @@ The following examples illustrate the behavior::
 Instant in time that includes the date and time of day with ``P`` digits of
 precision for the fraction of seconds and with a time zone. Values of this
 type are rendered using the time zone from the value.
-Time zones are expressed as the numeric UTC offset value::
+Time zones are expressed as the numeric UTC offset value or :doc:`IANA time zone
```
Comment (hashhar) [inline docs/src/main/sphinx/language/types.rst:242]: Should this be plain link instead of `doc`? Sphinx is complaining.

Comment (mosabua) [reply docs/src/main/sphinx/language/types.rst:242]: Yes, syntax is

```
`text <link>`_
```

Comment (findepi) [inline docs/src/main/sphinx/language/types.rst:242]: We also allow
- `UTC`
  - `GMT`, `Z`, `UT` as aliases for `UTC`
- `UTC+nn`, `UTC-nn` (or `UT` / `GMT` prefix) -- `nn` hour offset off of UTC


(see javadoc of `java.time.ZoneId#of(java.lang.String)`)

Comment (jhlodin) [reply docs/src/main/sphinx/language/types.rst:242]: Whoops. Fixing

**Comment by findepi** [comment]
The CI is red.

**Comment by mosabua** [comment]
@findepi I just checked out the branch locally and verified that the docs build works. The CI failure is misleading and caused by a flaky test from what I can tell.

**Comment by hashhar** [comment]
CI hit https://github.com/trinodb/trino/issues/13288

File: docs/src/main/sphinx/language/types.rst [hunk 3]
```diff
@@ -239,11 +239,22 @@ The following examples illustrate the behavior::
...
+* An `IANA time zone name <https://www.iana.org/time-zones>`_.
+
+The following examples demonstrate these syntax options::
+
+    SELECT TIMESTAMP '2001-08-22 03:04:05.321 UTC';
+    -- 2001-08-22 03:04:05.321 UTC
 
-    TIMESTAMP '2001-08-22 03:04:05.321 -08:00';
-    -- 2001-08-22 03:04:05.321-08:00
+    SELECT TIMESTAMP '2001-08-22 03:04:05.321 -08:00';
```
Comment (findepi) [inline docs/src/main/sphinx/language/types.rst:253]: Per above, this isn't an option. I.e. we need to improve the text above.

File: docs/src/main/sphinx/language/types.rst [hunk 4]
```diff
@@ -239,11 +239,22 @@ The following examples illustrate the behavior::
 Instant in time that includes the date and time of day with ``P`` digits of
 precision for the fraction of seconds and with a time zone. Values of this
 type are rendered using the time zone from the value.
-Time zones are expressed as the numeric UTC offset value::
+Time zones can be expressed in the following ways:
+
+* ``UTC``, with ``GMT``, ``Z``, or ``UT`` usable as aliases for UTC.
+* ``UTC +nn:00`` or ``UTC -nn:00`` with ``nn`` as an hour offset from UTC.
```
Comment (findepi) [inline docs/src/main/sphinx/language/types.rst:245]: The minutes part of offset doesn't have to be 0

```
trino> SELECT TIMESTAMP '2001-08-22 03:04:05.321 UTC+11:23';
             _col0
--------------------------------
 2001-08-22 03:04:05.321 +11:23
```

or, without "`UTC`"
```
trino> SELECT TIMESTAMP '2001-08-22 03:04:05.321 -08:12';
             _col0
--------------------------------
 2001-08-22 03:04:05.321 -08:12
```

**Comment by findepi** [comment]
thanks for updating https://github.com/trinodb/trino/pull/13927#discussion_r965835778 but it seems you dropped `UTC+11:23` option, it's now undocumented.

File: docs/src/main/sphinx/language/types.rst [hunk 5]
```diff
@@ -262,11 +262,26 @@ The following examples illustrate the behavior::
...
+    SELECT TIMESTAMP '2001-08-22 03:04:05.321 UTC';
+    -- 2001-08-22 03:04:05.321 UTC
+
+    SELECT TIMESTAMP '2001-08-22 03:04:05.321 -08:30';
+    -- 2001-08-22 03:04:05.321 -08:30
 
-    TIMESTAMP '2001-08-22 03:04:05.321 -08:00';
-    -- 2001-08-22 03:04:05.321-08:00
+    SELECT TIMESTAMP '2001-08-22 03:04:05.321 UTC -08:30';
+    -- 2001-08-22 03:04:05.321 UTC -08:30
```
Comment (findepi) [inline docs/src/main/sphinx/language/types.rst:281]: What does this query print when you run it in your Trino?

File: docs/src/main/sphinx/language/types.rst [hunk 6]
```diff
@@ -262,11 +262,26 @@ The following examples illustrate the behavior::
 Instant in time that includes the date and time of day with ``P`` digits of
 precision for the fraction of seconds and with a time zone. Values of this
 type are rendered using the time zone from the value.
-Time zones are expressed as the numeric UTC offset value::
+Time zones can be expressed in the following ways:
+
+* ``UTC``, with ``GMT``, ``Z``, or ``UT`` usable as aliases for UTC.
+* ``+hh:mm`` or ``-hh:mm`` with ``hh:mm``, with or without ``UTC`` or an alias,
```
Comment (findepi) [inline docs/src/main/sphinx/language/types.rst:268]: > an alias,

did you actually check which aliases work?

**Review by findepi — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/language/types.rst (hunk 6 above)
Comment (jhlodin) [reply docs/src/main/sphinx/language/types.rst:268]: UTC, GMT, and UT work as aliases with this syntax. Not Z.

File: docs/src/main/sphinx/language/types.rst (hunk 5 above)
Comment (jhlodin) [reply docs/src/main/sphinx/language/types.rst:281]: Thanks for pointing this out. The space after `UTC` is errant so it should just be `UTC-8:30` in the SELECT statement.

The output is `2001-08-22 03:04:05.321 -08:00` regardless of the UTC alias, Fixing.

File: docs/src/main/sphinx/language/types.rst (hunk 6 above)
Comment (mosabua) [reply docs/src/main/sphinx/language/types.rst]: Should we have tests for this in the codebase @findepi ?

**Review by findepi — APPROVED** [review]

File: docs/src/main/sphinx/language/types.rst [hunk 7]
```diff
@@ -262,11 +262,27 @@ The following examples illustrate the behavior::
...
-Time zones are expressed as the numeric UTC offset value::
+Time zones can be expressed in the following ways:
+
+* ``UTC``, with ``GMT``, ``Z``, or ``UT`` usable as aliases for UTC.
+* ``+hh:mm`` or ``-hh:mm`` with ``hh:mm`` as an hour and minute offset from UTC.
+  Can be written with or without ``UTC``, ``GMT``, or ``UT`` as an alias for
+  UTC.
+* An `IANA time zone name <https://www.iana.org/time-zones>`_.
+
+The following examples demonstrate these syntax options::
```
Comment (hashhar) [inline docs/src/main/sphinx/language/types.rst:273]: nit: the examples are not exhaustive list obviously

```suggestion
The following examples demonstrate some of these syntax options::
```

**Review by hashhar — APPROVED** [review]

File: docs/src/main/sphinx/language/types.rst (hunk 7 above)
Comment (jhlodin) [reply docs/src/main/sphinx/language/types.rst]: Good point, added

**Pull request merged**